lib/src/ast.dart

```diff
@@ -14,9 +14,7 @@ import 'package:path/path.dart' as path;
 import 'analyzer.dart';
 import 'utils.dart';
 
-List<String>? _reservedWords;
-
-List<String> get reservedWords => _reservedWords ??= _collectReservedWords();
+late final List<String> reservedWords = _collectReservedWords();
```

Comment:
You can (and I would) drop the `late` here. Top-level/static variables are inherently lazy, and always have been.

Reply:
Thanks!  This was absolutely a misunderstanding of `late` semantics on my part.  In the meantime, we added `unnecessary_late` to catch just such cases: https://github.com/dart-lang/linter/pull/3124.